### PR TITLE
RD-3587 Display number of executions on executionsStatus widget

### PR DESCRIPTION
**Pull request by NorbertAmartus**
## Description

To this point, `executionsStatus` widget was not displaying the number of executions, for each available chart.
Changes provided with this PR will alter that functionality, so that the mentioned number will be displayed.

### Before
Before

### After
After

## Checklist

- [x] My code follows the UI Code Style.
- [ ] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests

 

I've tested changes manually
Additionally, I've created test file for the `executionsStatus` widget and added a proper test scenario.

## Documentation

N/A

### CODE REVIEW

File: widgets/executionsStatus/src/widget.tsx [hunk 1]
```diff
@@ -1,16 +1,22 @@
 // @ts-nocheck File not migrated fully to TS
```
Comment (qooban) [inline widgets/executionsStatus/src/widget.tsx:1]: Maybe worth to remove this line and try migrating it fully? 🙂 

(you can take a look e.g. at `widgets/filters/src/widget.tsx` to see what kind of typings are expected for widget definition)

**Review by qooban — COMMENTED** [review]
Looks good 👍🏻 

Added just two comments indicating that maybe worth to migrate one file to TS (seems quick) and update test name (very quick).

File: test/cypress/integration/widgets/executions_status_spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+describe('ExecutionStatus widget', () => {
```
[the remark this answers is not in the document]
Comment (NorbertAmartus) [reply test/cypress/integration/widgets/executions_status_spec.ts:1]: Sure 😉

File: widgets/executionsStatus/src/widget.tsx (hunk 1 above)
Comment (NorbertAmartus) [reply widgets/executionsStatus/src/widget.tsx:1]: Hmm, okay
I'll try to do so :)

Comment (NorbertAmartus) [reply widgets/executionsStatus/src/widget.tsx]: Done 😉

File: widgets/executionsStatus/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// eslint-disable-next-line import/prefer-default-export
+
+export namespace ExecutionsStatusWidget {
+    export type Configuration = {
+        pollingTime: number;
+    };
+    export interface Params {
+        /* eslint-disable camelcase */
```
Comment (qooban) [inline widgets/executionsStatus/src/types.ts:8]: Optional:

It's not a big problem, but seeing `/* eslint-disable camelcase */` on one level and `/* eslint-enable camelcase */` on another level is a bit unexpected (at least for me). So alternatively we can disable it for the whole file.

File: widgets/executionsStatus/src/widget.tsx [hunk 4]
```diff
@@ -1,21 +1,27 @@
-// @ts-nocheck File not migrated fully to TS
```
Comment (qooban) [inline widgets/executionsStatus/src/widget.tsx:1]: Nicely done 👍🏻

File: widgets/common/src/UploadBlueprintForm.tsx [hunk 5]
```diff
@@ -128,7 +128,7 @@ class UploadBlueprintForm extends React.Component {
 
     resetErrors = () => {
         const { onChange } = this.props;
-        onChange(UploadBlueprintForm.NO_ERRORS);
+        onChange({ ...UploadBlueprintForm.NO_ERRORS });
```
Comment (qooban) [inline widgets/common/src/UploadBlueprintForm.tsx:131]: Oh, seems unrelated to this PR or am I missing something?
Is it a fix for RD-3733? 🙂

File: widgets/executionsStatus/src/types.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    interface DataItem {
+        executions: number;
+        status_display: string;
+    }
+
+    export interface Data {
+        items: DataItem[];
+    }
```
Comment (qooban) [inline widgets/executionsStatus/src/types.ts:22]: I think you can use `Stage.Types.PaginatedResponse`:
```suggestion
    interface DataItem {
        executions: number;
        status_display: string;
    }

    export type Data = Stage.Types.PaginatedResponse<DataItem>;
```
or even just:
```suggestion
    export type Data = Stage.Types.PaginatedResponse<{ executions: number; status_display: string }>;
```

**Review by qooban — APPROVED** [review]
Thanks for migrating one more file 🎉 

Added few more comments. None of the are real blockers.

File: widgets/executionsStatus/src/types.ts (hunk 3 above)
Comment (NorbertAmartus) [reply widgets/executionsStatus/src/types.ts:8]: Alright, I don't see a problem with that, so I'll do it right away :)

File: widgets/common/src/UploadBlueprintForm.tsx (hunk 5 above)
Comment (NorbertAmartus) [reply widgets/common/src/UploadBlueprintForm.tsx:131]: Yeah, you're right 😅
I've pushed one commit into the incorrect branch and it seems that I didn't remove the last bit of those changes

Great catch! 💪

File: widgets/executionsStatus/src/types.ts (hunk 3 above)
Comment (NorbertAmartus) [reply widgets/executionsStatus/src/types.ts]: Done 😉

File: widgets/common/src/UploadBlueprintForm.tsx (hunk 5 above)
Comment (NorbertAmartus) [reply widgets/common/src/UploadBlueprintForm.tsx]: Original line has been restored 😁

File: widgets/executionsStatus/src/types.ts (hunk 6 above)
Comment (NorbertAmartus) [reply widgets/executionsStatus/src/types.ts:22]: Whoa!
Now I don't know why should I do that! 😁 

Thanks for sending that type, looks very handy!
I'll implement that right away!

Comment (NorbertAmartus) [reply widgets/executionsStatus/src/types.ts]: Done 😎

File: widgets/executionsStatus/src/types.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line import/prefer-default-export
```
Comment (qooban) [inline widgets/executionsStatus/src/types.ts:1]: It seems that this line is not necessary or am I missing something?

**Review by qooban — DISMISSED** [review]
Looks good! 👍🏻 

I've just found one unnecessary line, can be fixed at once or by the way within another PR.

File: widgets/executionsStatus/src/types.ts (hunk 7 above)
Comment (NorbertAmartus) [reply widgets/executionsStatus/src/types.ts:1]: Yup, you're right 😁 
I'll remove it within this PR, so that we may have git history only related to this jira ticket 😉

Comment (NorbertAmartus) [reply widgets/executionsStatus/src/types.ts]: Done 💪

**Review by qooban — APPROVED** [review]
